app/models/tax_return.rb

```diff
@@ -23,4 +24,94 @@
+      FINALIZE => I18n.t("case_management.tax_returns.stage.final_steps"),
+      FILE => I18n.t("case_management.tax_returns.stage.filing_completed"),
+  }.freeze
+
+  enum status: {
+      intake_in_progress: 101, intake_open: 102, intake_review: 103, intake_more_info: 104, intake_info_requested: 105, intake_needs_assignment: 106,
+      prep_ready_for_call: 201, prep_more_info: 202, prep_preparing: 203, prep_ready_for_review: 204,
+      review_in_review: 301, review_complete_signature_requested: 302, review_more_info: 303,
+      finalize_closed: 401, finalize_signed: 402,
+      filed_e_file: 501, filed_mail_file: 502, filed_rejected: 503, filed_accepted: 504
+  }, _prefix: :status
+
+  STATUSES =
+      {
+          INTAKE => {
+              "intake_in_progress" => I18n.t("case_management.tax_returns.status.in_progress"),
+              "intake_open" => I18n.t("case_management.tax_returns.status.open"),
+              "intake_review" => I18n.t("case_management.tax_returns.status.in_review"),
+              "intake_more_info" => I18n.t("case_management.tax_returns.status.needs_more_information"),
+              "intake_info_requested" => I18n.t("case_management.tax_returns.status.info_requested"),
+              "intake_needs_assignment" => I18n.t("case_management.tax_returns.status.needs_assignment"),
+          },
+          PREP => {
+              "prep_ready_for_call" => I18n.t("case_management.tax_returns.status.ready_for_call"),
+              "prep_more_info" => I18n.t("case_management.tax_returns.status.needs_more_information"),
+              "prep_preparing" => I18n.t("case_management.tax_returns.status.entering_in_tax_slayer"),
+              "prep_ready_for_review" => I18n.t("case_management.tax_returns.status.ready_for_qr"),
+          },
+          REVIEW => {
+              "review_in_review" => I18n.t("case_management.tax_returns.status.in_review"),
+              "review_complete_signature_requested" => I18n.t("case_management.tax_returns.status.complete_signature_requested"),
+              "review_more_info" => I18n.t("case_management.tax_returns.status.needs_more_information"),
+          },
+          FINALIZE => {
+              "finalize_closed" => I18n.t("case_management.tax_returns.status.closed"),
+              "finalize_signed" => I18n.t("case_management.tax_returns.status.return_signed"),
+          },
+          FILE => {
+              "filed_e_file" => I18n.t("case_management.tax_returns.status.efiled"),
+              "filed_mail_file" => I18n.t("case_management.tax_returns.status.mail_filed"),
+              "filed_rejected" => I18n.t("case_management.tax_returns.status.rejected"),
+              "filed_accepted" => I18n.t("case_management.tax_returns.status.accepted"),
+          },
+      }.freeze
+
+  def self.stages
+    STAGES
+  end
+
+  def self.statuses_for(stage_key)
+    STATUSES[stage_key]
+  end
+
+  def self.grouped_status_options
```

Comment:
There seems to be an odd caching thing happening here - when I switch locales the options in the select stay the same but when I put a breakpoint in this method they changed. If no one has a suggestion I may try moving this into a view helper next.

Reply:
I think this makes more sense as a view helper anyway.

Reply:
refactored into a helper.